_docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/WordpressApi.php

```diff
@@ -96,36 +145,49 @@ public function getContentByCategory($categories, $max_results, $select_logic) {
     $query['per_page'] = $max_results;
     $query['logic'] = $select_logic_tmp;
 
-    if ($categories) {
-      if (count($categories) > 0) {
-        $query['categories'] = implode(",", $categories);
-      }
+    if (!empty($categories)) {
+      $query['categories'] = implode(",", $categories);
+      $imploded_categories = implode('', $categories);
+      $cid = "wordpress_api:by_category:per_page_{$query['per_page']}-logic_{$query['logic']}-categories_{$imploded_categories}";
+    }
+    else {
+      $cid = "";
     }
 
-    try {
-      // Retrieve the WP posts from the $feed_url and decode the JSON into a
-      // $results array.
-      $request = $this->client->request('GET', $feed_url, ['query' => $query]);
-
-      $response = $request->getBody()->getContents();
+    // Serve the cached Wordpress API data if available.
+    if ($cache = $this->cacheBin->get($cid)) {
+      $data = $cache->data;
+    }
+    // If this is uncached or the cache is invalid, fetch fresh data.
+    else {
+      try {
+        // Retrieve the WP posts from the $feed_url and decode the JSON into a
+        // $results array.
+        $request = $this->client->request('GET', $feed_url, ['query' => $query]);
 
-      $results = json_decode($response);
+        $response = $request->getBody()->getContents();
 
-      // Pass the WP posts returned in $results, and get the processed/formatted
-      // results from getContentCompositeMultiple().
-      $items = $this->getContentCompositeMultiple($results);
+        $results = json_decode($response);
 
-      return $items;
-    }
-    catch (\Exception $e) {
-      \Drupal::logger('rhd_assemblies')->error(
-        "Exception while calling Wordpress API: @message", [
-          '@message' => $e->getMessage(),
-        ]
-      );
-
-      return [];
+        // Pass the WP posts returned in $results, and get the
+        // processed/formatted results from getContentCompositeMultiple().
+        $data = $this->getContentCompositeMultiple($results);
```

Comment:
Do you want to address the 9 php warnings caused by this method when there is no data? See RHDX-169